### PR TITLE
Add format to performer field placeholder

### PR DESCRIPTION
**Pull request by UncleRoger33**
There are various birthday formats and unless you use YYYY-MM-DD format, the age calculation function don't work on performers page. 
It's not mentioned on Model Edit page text field.

### CODE REVIEW

**Review by peolic — APPROVED** [review]

**Comment by bnkai** [comment]
In that case shouldnt we use "Birthdate (YYYY-MM-DD)" ?
A plain YYYY-MM-DD feels a bit weird imho.

**Comment by peolic** [comment]
I thought about it as well, but there is a "Birthdate" label right next to it.

**Comment by bnkai** [comment]
date

**Comment by peolic** [comment]
Yeah in that case I would suggest adding a `placeholder` optional parameter to [`renderTextField`](https://github.com/stashapp/stash/blob/develop/ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx#L728-L744):
```diff
- function renderTextField(field: string, title: string) {
+ function renderTextField(field: string, title: string, placeholder?: string) {
    return (
      <Form.Group controlId={field} as={Row}>
        <Form.Label column xs={labelXS} xl={labelXL}>
          {title}
        </Form.Label>
        <Col xs={fieldXS} xl={fieldXL}>
          <Form.Control
            className="text-input"
-           placeholder={title}
+           placeholder={placeholder ?? title}
            {...formik.getFieldProps(field)}
            isInvalid={!!formik.getFieldMeta(field).error}
          />
        </Col>
      </Form.Group>
    );
  }
```
and
```ts
{renderTextField("birthdate", "Birthdate", "YYYY-MM-DD")}
```

**Comment by UncleRoger33** [comment]
> 
> 
> Yeah in that case I would suggest adding a `placeholder` optional parameter to [`renderTextField`](https://github.com/stashapp/stash/blob/develop/ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx#L728-L744):
> 
> ```diff
> - function renderTextField(field: string, title: string) {
> + function renderTextField(field: string, title: string, placeholder?: string) {
>     return (
>       <Form.Group controlId={field} as={Row}>
>         <Form.Label column xs={labelXS} xl={labelXL}>
>           {title}
>         </Form.Label>
>         <Col xs={fieldXS} xl={fieldXL}>
>           <Form.Control
>             className="text-input"
> -           placeholder={title}
> +           placeholder={placeholder ?? title}
>             {...formik.getFieldProps(field)}
>             isInvalid={!!formik.getFieldMeta(field).error}
>           />
>         </Col>
>       </Form.Group>
>     );
>   }
> ```
> 
> and
> 
> ```ts
> {renderTextField("birthdate", "Birthdate", "YYYY-MM-DD")}
> ```

Yep, this looks better 👍 

@bnkai already posted a screenshot but here is mine (with today's latest release)
"1995/10/27", "27-10-1995", "Oct 27 1995", etc. don't work and i found the right YYYY-MM-DD format in a community scraper script. A bit confusing as it is now.


stash

**Comment by WithoutPants** [comment]
> Yep, this looks better 👍

Can you update the branch with this change?

**Comment by UncleRoger33** [comment]
> 
> 

> 
> Can you update the branch with this change?

Done.

File: ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx [hunk 1]
```diff
@@ -725,7 +725,7 @@ export const PerformerEditPanel: React.FC<IPerformerDetails> = ({
     );
   }
 
-  function renderTextField(field: string, title: string) {
+function renderTextField(field: string, title: string, placeholder?: string) {
```
Comment (peolic) [inline ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx:728]:
```suggestion
  function renderTextField(field: string, title: string, placeholder?: string) {
```

to appease prettier (and fix the build)

Comment (UncleRoger33) [reply ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx]: 👍

**Review by peolic — APPROVED** [review]
Looks good now 👍 

image

**Comment by UncleRoger33** [comment]
> 
> 
> Looks good now 👍
> 
> image

Yep, that's what i was looking for :)

**Pull request merged**